kedro-agentic-workflows/src/kedro_agentic_workflows/datasets/langfuse_trace_dataset.py

```diff
@@ -0,0 +1,230 @@
+        ```
+
+        Using Python API:
+
+        ```python
+        from kedro_agentic_workflows.datasets.langfuse_trace_dataset import LangfuseTraceDataset
+
+        # Basic usage
+        dataset = LangfuseTraceDataset(
+            credentials={
+                "public_key": "pk_...", 
+                "secret_key": "sk_...", 
+                "host": "https://cloud.langfuse.com",
+                "openai": {"openai_api_key": "sk-..."}
+            },
+            mode="openai"
+        )
+
+        # Load tracing client
+        client = dataset.load()
+        response = client.chat.completions.create(...)  # Automatically traced
+        ```
+    """
+
+    def __init__(
+        self,
+        credentials: dict[str, Any],
+        mode: Literal["langchain", "openai", "sdk"] = "sdk",
+        **trace_kwargs
+    ):
+        """Initialize LangfuseTraceDataset and configure environment variables.
+        
+        Validates credentials and sets up appropriate environment variables for
+        Langfuse tracing integration. Environment variables are set immediately
+        during initialization for use by all tracing modes.
+        
+        Args:
+            credentials: Dictionary with Langfuse credentials. Required keys:
+                {public_key, secret_key, host}. For OpenAI mode, may also include
+                openai section with {openai_api_key, openai_api_base}.
+            mode: Tracing mode - "langchain", "openai", or "sdk" (default).
+            **trace_kwargs: Additional kwargs passed to the tracing client.
+            
+        Raises:
+            ValueError: If required Langfuse credentials are missing or empty.
+            
+        Examples:
+            >>> # Basic SDK mode
+            >>> dataset = LangfuseTraceDataset(
+            ...     credentials={"public_key": "pk_...", "secret_key": "sk_...", "host": "..."}
+            ... )
+            
+            >>> # OpenAI mode with API key
+            >>> dataset = LangfuseTraceDataset(
+            ...     credentials={
+            ...         "public_key": "pk_...", "secret_key": "sk_...", "host": "...",
+            ...         "openai": {"openai_api_key": "sk-...", "openai_api_base": "..."}
+            ...     },
+            ...     mode="openai"
+            ... )
+            
+        Note:
+            Sets LANGFUSE_SECRET_KEY, LANGFUSE_PUBLIC_KEY, and LANGFUSE_HOST 
+            environment variables from the provided credentials. Also sets 
+            OPENAI_API_KEY if provided for OpenAI mode compatibility.
+        """
+        self._credentials = credentials
+        self._mode = mode
+        self._trace_kwargs = trace_kwargs
+        
+        # Validate Langfuse credentials before setting environment variables
+        self._validate_langfuse_credentials()
+        
+        # Set Langfuse environment variables from credentials
+        os.environ["LANGFUSE_SECRET_KEY"] = self._credentials.get("secret_key", "")
```

Comment:
Since we already validated them we can do the following:

```suggestion
        os.environ["LANGFUSE_SECRET_KEY"] = self._credentials["secret_key"]
        os.environ["LANGFUSE_PUBLIC_KEY"] = self._credentials["public_key"]
        if "host" in self._credentials:
            os.environ["LANGFUSE_HOST"] = self._credentials["host"]
```